crates/language/src/buffer_tests.rs

```diff
@@ -3140,6 +3141,81 @@ fn test_trailing_whitespace_ranges(mut rng: StdRng) {
     );
 }
 
+#[gpui::test]
+fn test_words_in_range(cx: &mut gpui::App) {
+    init_settings(cx, |_| {});
+
+    let contents = r#"let word=foo.bar Foo word2-FoO-Pizza-word FOO word"#;
+
+    let buffer = cx.new(|cx| {
+        let buffer = Buffer::local(contents, cx).with_language(Arc::new(rust_lang()), cx);
+        assert_eq!(buffer.text(), contents);
+        buffer.check_invariants();
+        buffer
+    });
+
+    buffer.update(cx, |buffer, _| {
```

Comment:
I think in this test, it would probably be good to have some assertions where the query (and the buffer) contain multibyte characters.

Reply:
Done, and that lead to a better way to compare chars case-insensitively, thank you.